storybookjs/storybook
# CLI: Add url flag to download templates in sb repro

Issue: N/A

## What I did

This PR integrates `degit` into the sb cli.

It adds two features:

### Prefer downloading from storybook repro repository

This way, the CLI just downloads a git repo from [repro-templates](https://github.com/storybookjs/repro-templates) instead of having to run a generator (e.g. create-react-app myApp...). This could save time both in CI but also for the `sb repro` experience.

If the selected template does not exist in repro-templates, it will fallback to the original generator.

For the user, there's no change in the command. Optionally, you could use the `--skip-git` flag to use the generators directly instead.

The repro template has both `main` and `next` branches which old latest vs next versions of Storybook. The script will detect whether the user is using `sb repro` or `sb repro@some-alpha-version` and will download from the correct branch

### Set project via --url flag

`sb repro` now has a `--url` flag which allows you to download any git project (github, gitlab, bitbucket..) and bootstrap Storybook on top of it. Could be useful for QAing purposes in case we find big projects with certain settings that could be valuable to test out. 

This uses `degit` under the hood so you can use all kinds of formats for the URL such as `username/project` or `https://github.com/username/project` or other formats that degit supports.

## How to test

First check the branch, bootstrap it or just build the cli:
```
yarn build cli
```

Next, somewhere in an experimentation folder, run `sb repro` using the URL of a project that does not contain Storybook.

```
# This will prompt for location folder, using the repo as default folder name
../storybook/lib/cli/bin/index.js repro --url the-url-here
```

```
# This will download and install things automatically with no prompt
../storybook/lib/cli/bin/index.js repro --url the-url-here outputfoldername
```


If your answer is yes to any of these, please make sure to include it in your PR.
